response.go

```diff
@@ -22,7 +22,7 @@ func NewResponse() Response {
 	return Response{
 		Meta:       map[string]interface{}{},
 		Header:     http.Header{},
-		StatusCode: 200,
+		StatusCode: -1,
```

Comment:
I don't particularly like setting this to `-1` so that I can detect it later, I'd prefer to just not have a value and compare it to `nil` but I think that's the JavaScript side of me talking.

Would love to know if there is a better way to do this in Go.

Reply:
Could also not set it and check against 0. The style though I think is fine. I can't think of a particularly better way of doing it while needing to test for user override. Just requires care to make sure -1 or whatever cant make it all the way down.

Reply:
Actually, if the status function chains and returns an altered context you wont even need the -1 or the checks.  It can stay 200 as the default and just altered and passed back if status is called.

```go
func (c Context) Status(code int) Context {
     c.Response.StatusCode = code
     return c
}
```

Which would give you your
```go
return c.Status(201).Respond(data)
return c.Status(400).Error(err)
```

Reply:
Agreed, that would definitely make it easier and give us a simpler interface looking ahead towards needing the ability to do a response interceptor for Gzip (I'll get my branch for that updated too with these changes)